### PR TITLE
Bump worker ignition specs to 3.4

### PR DESCRIPTION
**Pull request by jrvaldes**
This change updates the ignition specs given the previous version (3.2) does not support the parsing of the 3.4 spec

No backports required given "[4.13 MCO will always render configs with 3.2.0](https://redhat-internal.slack.com/archives/CM4ERHBJS/p1692039942242409?thread_ts=1692035137.053939&cid=CM4ERHBJS)"

### CODE REVIEW

**Review by sebsoto — APPROVED** [review]

File: pkg/ignition/ignition.go
Comment (aravindhp) [inline pkg/ignition/ignition.go:1]: Please fix the commit message:
> This change updates the ignition specs given the previous version (3.2) does not support the parsing of the 3.4 spec

**Review by aravindhp — APPROVED** [review]
Thanks for fixing this, @jrvaldes

File: pkg/ignition/ignition.go
Comment (jrvaldes) [reply pkg/ignition/ignition.go:1]: Good catch, will do after the `gcp-e2e-operator` job succeed  🤞🏼

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm, can re-apply after commit msg is updated

**Comment by jrvaldes** [comment]
/test vsphere-e2e-operator

 gcp-e2e-operator job got stuck at `Acquiring leases for test gcp-e2e-operator` step 

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1728/pull-ci-openshift-windows-machine-config-operator-master-gcp-e2e-operator/1691161029381197824

**Comment by jrvaldes** [comment]
ipi-install step  [failed](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1728/pull-ci-openshift-windows-machine-config-operator-master-vsphere-e2e-operator/1691179883121610752)

**Comment by jrvaldes** [comment]
the `gcp-e2e-operator ` [failed](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1728/pull-ci-openshift-windows-machine-config-operator-master-gcp-e2e-operator/1691439575076966400), however the ignition version was successfully parsed and a Windows node came up.


```
2023-08-15T15:00:20Z	DEBUG	ignition	parsed	{"machineconfig": "rendered-worker-13ad43a1262d5f01d3f7cd24e4f8cc54", "ignition version": "3.4.0"}

...

2023-08-15T15:01:05Z	INFO	controllers.configmap	instance is up to date	{"node": "ci-op-z3s064xm-8411c-xq5vn-e2e-j6jvt", "version": "9.0.0-ce58257-dirty"}
2023-08-15T15:01:05Z	DEBUG	events	Configured instance with address 10.0.128.6 as a worker node	{"type": "Normal", "object": {"kind":"ConfigMap","namespace":"alternate-wmco-namespace","name":"windows-instances","apiVersion":"v1"}, "reason": "InstanceSetup"}

```

See WMCO [logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1728/pull-ci-openshift-windows-machine-config-operator-master-gcp-e2e-operator/1691439575076966400/artifacts/gcp-e2e-operator/gather-extra/artifacts/pods/alternate-wmco-namespace_windows-machine-config-operator-5bbdc87b6b-8qsnv_manager.log) for the given job.

**Comment by aravindhp** [comment]
@jrvaldes the log you have posted implies the instance was configured at some point. But why is it failing reconfiguration? Looks like SSH connection is failing:

```
2023-08-15T15:22:17Z	ERROR	Reconciler error	{"controller": "machine", "controllerGroup": "machine.openshift.io", "controllerKind": "Machine", "Machine": {"name":"ci-op-z3s064xm-8411c-xq5vn-e2e-wm-gvfvk","namespace":"openshift-machine-api"}, "namespace": "openshift-machine-api", "name": "ci-op-z3s064xm-8411c-xq5vn-e2e-wm-gvfvk", "reconcileID": "d980e0a9-44fc-4678-b799-ffc62d94e7d4", "error": "unable to configure instance ci-op-z3s064xm-8411c-xq5vn-e2e-wm-gvfvk: failed to create new nodeconfig: error instantiating Windows instance from VM: unable to setup VM 10.0.128.7 sshConnectivity: error instantiating SSH client: unable to connect to Windows VM 10.0.128.7: timed out waiting for the condition"}
```
I don't see anything in the machine-api-controller [logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1728/pull-ci-openshift-windows-machine-config-operator-master-gcp-e2e-operator/1691439575076966400/artifacts/gcp-e2e-operator/gather-extra/artifacts/pods/openshift-machine-api_machine-api-controllers-fcd9cd4db-b75j2_machine-controller.log)

**Comment by jrvaldes** [comment]
> But why is it failing reconfiguration?

I did not look deeper. 

@openshift/openshift-team-windows-containers  The point I'm making is that the proposed changes were validated; worker ignition was parsed, and a Windows node came up.

**Comment by aravindhp** [comment]
/override ci/prow/gcp-e2e-operator ci/prow/nutanix-e2e-operator

**Comment by aravindhp** [comment]
/override ci/prow/azure-e2e-operator

The test passed previously

**Comment by aravindhp** [comment]
/override ci/prow/vsphere-proxy-e2e-operator

The test passed previously

**Pull request merged**